Return-To-The-Roots/s25client
# Minor bugfixes and increase test coverage to 100%

Test code should be fully covered to
1) Make sure all intended code is run
2) Predicate checks work as expected

- use LCOV comments to mark unreachable code (fallbacks)
- fix bug in smartBmp test
- test predicates and output
- some bugfixes and preparation for bow delivery of #1056
